src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTest.java

```diff
@@ -0,0 +1,163 @@
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class RangeQueryBuilderTest extends BaseQueryTestCase<RangeQueryBuilder> {
+
+    private static final List<String> TIMEZONE_IDS = new ArrayList<>(DateTimeZone.getAvailableIDs());
+
+    @Override
+    protected RangeQueryBuilder createTestQueryBuilder() {
+        RangeQueryBuilder query = new RangeQueryBuilder(randomAsciiOfLengthBetween(1, 10));
+        // switch between numeric and date ranges
+        if (randomBoolean()) {
+            if (randomBoolean()) {
+                query = new RangeQueryBuilder(INT_FIELD_NAME);
+                query.from(randomIntBetween(1, 100));
+                query.to(randomIntBetween(101, 200));
+            } else {
+                query.from(0.0-randomDouble());
+                query.to(randomDouble());
+            }
+        } else {
+            query = new RangeQueryBuilder(DATE_FIELD_NAME);
+            query.from(new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000)).toString());
+            query.to(new DateTime(System.currentTimeMillis() + randomIntBetween(0, 1000000)).toString());
+            if (randomBoolean()) {
+                query.timeZone(TIMEZONE_IDS.get(randomIntBetween(0, TIMEZONE_IDS.size()-1)));
+            }
+            if (randomBoolean()) {
+                query.format("yyyy-MM-dd'T'HH:mm:ss.SSSZZ");
+            }
+        }
+        query.includeLower(randomBoolean()).includeUpper(randomBoolean());
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLengthBetween(1, 10));
+        }
+
+        if (randomBoolean()) {
+            query.from(null);
+        }
+        if (randomBoolean()) {
+            query.to(null);
+        }
+        return query;
+    }
+
+    @Override
+    protected void assertLuceneQuery(RangeQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+        String fieldname = queryBuilder.fieldname();
+        if (!fieldname.equals(DATE_FIELD_NAME) && !fieldname.equals(INT_FIELD_NAME)) {
+            assertThat(query, instanceOf(TermRangeQuery.class));
+            TermRangeQuery termRangeQuery = (TermRangeQuery) query;
+            assertThat(termRangeQuery.includesLower(), is(queryBuilder.includeLower()));
+            assertThat(termRangeQuery.includesUpper(), is(queryBuilder.includeUpper()));
+            assertThat(termRangeQuery.getLowerTerm(), is(BytesRefs.toBytesRef(queryBuilder.from())));
+            assertThat(termRangeQuery.getUpperTerm(), is(BytesRefs.toBytesRef(queryBuilder.to())));
+            assertThat(termRangeQuery.getField(), is(fieldname));
```

Comment:
I wonder if it makes as best practice to always create the expected lucene query and rely on lucene's equals rather than going through each field. Thoughts?